src/handlers.cpp

```diff
@@ -385,6 +385,12 @@ void ImpiTask::on_mar_response(Diameter::Message& rsp)
       }
     }
     break;
+    case DIAMETER_UNABLE_TO_DELIVER:
+      // LCOV_EXCL_START - nothing interesting to UT.
+      TRC_ERROR("Unable to deliver MAR - suggest retrying from a different Homestead node");
```

Comment:
I'm not sure this log is useful - I generally find https://github.com/Metaswitch/cpp-common/blob/master/src/diameterstack.cpp#L288 enough to diagnose this issue.

What it might be worth having here is a comment recording why this is the right ting to do (matching our IM discussion - i.e. another Homestead might have a connection to the HSS, or a connection to an SLF which has a connection to the HSS, and returning 503 means Sprout might try that Homestead).